Assignment6/Array.cs

```diff
@@ -1,10 +1,91 @@
 ﻿using System;
+using System.Collections;
 using System.Collections.Generic;
 
+#pragma warning disable CS8653
+#pragma warning disable CS8602
+#pragma warning disable CA2208
+#pragma warning disable CA1303
+
 namespace Assignment6
 {
-    public class Array<T>
+    public class ArrayCollection<T> : ICollection<T>
     {
-        
+        public int Count => _Data.Count;

```

Comment:
```suggestion
        public int Count { get => _Data.Count; }
```
Read-only properties will prevent users of this class from modifying this value, only allowing retrieval.  This also  grabs the current value of `_Data.Count` when the property is requested, whereas the current `public int Count => _Data.Count;` only gets the value of `_Data.Count` once, at instantiation.